src/ray/core_worker/transport/direct_actor_task_submitter.cc

```diff
@@ -580,12 +580,21 @@ void CoreWorkerDirectActorTaskSubmitter::HandlePushTaskReply(
       // If the actor is already dead, immediately mark the task object as failed.
       // Otherwise, start the grace period before marking the object as dead.
       is_actor_dead = queue.state == rpc::ActorTableData::DEAD;
-      const auto &death_cause = queue.death_cause;
-      error_info = GetErrorInfoFromActorDeathCause(death_cause);
-      error_type = error_info.error_type();
-      fail_immediatedly = error_info.has_actor_died_error() &&
-                          error_info.actor_died_error().has_oom_context() &&
-                          error_info.actor_died_error().oom_context().fail_immediately();
+      if (is_actor_dead) {
+        const auto &death_cause = queue.death_cause;
+        error_info = GetErrorInfoFromActorDeathCause(death_cause);
+        error_type = error_info.error_type();
+        fail_immediately = error_info.has_actor_died_error() &&
+                           error_info.actor_died_error().has_oom_context() &&
+                           error_info.actor_died_error().oom_context().fail_immediately();
+      } else {
+        // actor is not dead, but the request failed. Consider it as a temporal one.
```

Comment:
Does this case catch all transient, unexpected system failures that cause requests to fail? Or is it specific to network failures only?

```suggestion
        // actor may or may not be dead, but the request failed. Consider the failure temporary.
```